src/global.rs

```diff
@@ -130,30 +132,52 @@ impl<T> Default for Global<T> {
 
 impl<T> Drop for Global<T> {
   fn drop(&mut self) {
-    if !self.is_empty() {
-      panic!("Global handle dropped while holding a value");
+    match &mut self.value {
+      None => {
+        // This global handle is empty.
+        assert!(self.isolate_handle.is_none())
+      }
+      Some(_)
+        if unsafe {
+          self
+            .isolate_handle
+            .as_ref()
+            .unwrap()
+            .get_isolate_ptr()
+            .is_null()
+        } =>
+      {
+        // This global handle is associated with an Isolate that has already
+        // been disposed.
+      }
+      addr @ Some(_) => unsafe {
+        // Destroy the storage cell that contains the contents of this Global.
+        v8__Global__Reset__0(
+          &mut *(addr as *mut Option<NonNull<T>> as *mut *mut Value),
+        )
```

Comment:
If the Isolate was destroyed already, but this one wasn't, wouldn't we have gotten a segfault? I think this is unreachable.

Reply:
??

Reply:
* Check out https://github.com/denoland/rusty_v8/tree/proof
* Run `cargo test global -- --test-threads=1 --nocapture`